Sources/MapboxMaps/Foundation/ResourceOptionsManager.swift

```diff
@@ -0,0 +1,147 @@
+    ///
+    /// You can override the shared global access token, using
+    /// `ResourceOptionsManager.default`:
+    ///
+    ///     ```
+    ///     ResourceOptionsManager.default.resourceOptions.accessToken = "overridden-access-token"
+    ///     ```
+    ///
+    /// - Parameter accessToken: Valid access token or `nil`
+    public convenience init(accessToken: String? = nil) {
+        let resourceOptions = ResourceOptions(accessToken: accessToken ?? "")
+        self.init(resourceOptions: resourceOptions)
+    }
+
+    public convenience init(resourceOptions: ResourceOptions) {
+        self.init(resourceOptions: resourceOptions, for: .main)
+    }
+
+    /// Initializes a `ResourceOptionsManager` with the specified access token
+    /// and `Bundle`.
+    ///
+    /// - Parameters:
+    ///     - accessToken: Valid access token or `nil`
+    ///     - bundle: Bundle to search for an access token (used if resource
+    ///         options is nil).
+    internal init(resourceOptions: ResourceOptions, for bundle: Bundle) {
+        self.bundle = bundle
+        update(resourceOptions)
+    }
+
+    /// Reset the manager to a default `ResourceOptions` using the specified
+    /// access token.
+    ///
+    /// - Parameter accessToken: Valid access token or `nil`
+    internal func reset(resourceOptions: ResourceOptions? = nil) {
+        self.resourceOptions = resourceOptions ?? ResourceOptions(accessToken: defaultAccessToken())
+    }
+
+    private func update(_ resourceOptions: ResourceOptions) {
+        // Update access token
+        let token = resourceOptions.accessToken.isEmpty ?
+            defaultAccessToken() : resourceOptions.accessToken
+
+        _resourceOptions = resourceOptions
+        _resourceOptions.accessToken = token
+
+        let resolvedTileStore = resourceOptions.tileStore ?? TileStore.default
+        if resourceOptions.tileStoreUsageMode != .disabled {
+            resolvedTileStore.setAccessToken(token)
+        }
+
+        tileStore = resolvedTileStore
+    }
+
+    internal func defaultAccessToken() -> String {
+        // Check User defaults
+        #if DEBUG
+        if let accessToken = UserDefaults.standard.string(forKey: "MBXAccessToken") {
+            print("Found access token from UserDefaults (command line parameter?)")
+            return accessToken
+        }
+        #endif
+
+        var token = ""
+
+        // Check application plist
+        if let accessToken = bundle.infoDictionary?["MBXAccessToken"] as? String {
+            token = accessToken
+        }
+        // Check for a bundled file
+        else if let url = bundle.url(forResource: "MapboxAccessToken", withExtension: nil),
+                let tokenFromFile = try? String(contentsOf: url) {
+            token = tokenFromFile
+        }
+
+        if token.isEmpty {
+            Log.warning(forMessage: "Empty access token.", category: "ResourceOptions")
+        }
```

Comment:
So you don't throw exceptions when the token is not found, and instead just show a warning message? What does it look like to users? will the MapView will show a black screen?

Reply:
fyi, if the access token is not found anywhere, we will throw a runtime exception that crashes the app..

cc @Chaoba

Reply:
We could do the same - though there's some trickiness around testing.